src/services/analytics/tx-tracking.ts

```diff
@@ -0,0 +1,68 @@
+import { TX_TYPES } from '@/services/analytics/events/transactions'
+import { getTxDetails } from '@/services/tx/txDetails'
+import { isWalletConnectSafeApp } from '@/services/walletconnect/utils'
+import { SettingsInfoType, type TransactionDetails } from '@safe-global/safe-gateway-typescript-sdk'
+import {
+  isERC721Transfer,
+  isMultiSendTxInfo,
+  isSettingsChangeTxInfo,
+  isTransferTxInfo,
+  isCustomTxInfo,
+  isCancellationTxInfo,
+} from '@/utils/transaction-guards'
+
+export const getTransactionTrackingType = async (txId: string, chainId: string): Promise<string | undefined> => {
+  let details: TransactionDetails
+  try {
+    details = await getTxDetails(txId, chainId)
+  } catch {
+    return TX_TYPES.custom
+  }
+  const { txInfo } = details
+
+  if (isTransferTxInfo(txInfo)) {
+    if (isERC721Transfer(txInfo.transferInfo)) {
+      return TX_TYPES.transfer_nft
+    }
+    return TX_TYPES.transfer_token
```

Comment:
Would it be interesting for us to differentiate between ERC20 and native transfers?

Reply:
@alicecha wdyt?